### PR TITLE
feat(auth): customer mover script

### PR DESCRIPTION
**Pull request by julianpoy**
## Because

* We need to be able to migrate customers off of a given plan.

## This pull request

* Adds a script to move customers from one plan to another.

## Issue that this pull request solves

Closes FXA-7125
Closes FXA-7122

### CODE REVIEW

File: packages/fxa-auth-server/scripts/move-customers-to-new-plan.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    .version(pckg.version)
+    .option(
+      '-b, --batch-size [number]',
+      'Number of subscriptions to query from firestore at a time.  Defaults to 100.',
+      100
+    )
+    .option(
+      '-o, --output-file [string]',
+      'Output file to write report to. Will be output in CSV format.  Defaults to stripe-tax-existing-customers-output.csv.',
+      'move-customers-to-new-plan-output.csv'
```
[the remark this answers is not in the document]
Comment (julianpoy) [reply packages/fxa-auth-server/scripts/move-customers-to-new-plan.ts:34]: Yep that's a typo

File: packages/fxa-auth-server/scripts/move-customers-to-new-plan/move-customers-to-new-plan.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+   * @param database A reference to the FXA database
+   */
+  constructor(
+    private sourcePlanId: string,
+    private destinationPlanId: string,
+    private excludePlanIds: string[],
+    private batchSize: number,
+    private outputFile: string,
+    private stripeHelper: StripeHelper,
+    rateLimit: number,
```
[the remark this answers is not in the document]
Comment (julianpoy) [reply packages/fxa-auth-server/scripts/move-customers-to-new-plan/move-customers-to-new-plan.ts:49]: It's not used anywhere in the function - only in the constructor so not needed

File: packages/fxa-auth-server/scripts/move-customers-to-new-plan/move-customers-to-new-plan.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    private database: any
```
[the remark this answers is not in the document]
Comment (julianpoy) [reply packages/fxa-auth-server/scripts/move-customers-to-new-plan/move-customers-to-new-plan.ts:50]: Not from what I saw. I don't think we type the FXA database at the moment but I could be mistaken.

File: packages/fxa-auth-server/scripts/move-customers-to-new-plan/move-customers-to-new-plan.ts [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+   * @param startAfter ID of the last element of the previous batch for pagination
+   * @returns A list of subscriptions from firestore
+   */
+  async fetchSubsBatch(startAfter: string | null) {
+    const collectionPrefix = `${this.config.authFirestore.prefix}stripe-`;
+    const subscriptionCollection = `${collectionPrefix}subscriptions`;
+
+    const subscriptionSnap = await this.firestore
+      .collectionGroup(subscriptionCollection)
+      .where('plan.id', '==', this.sourcePlanId)
```
Comment (bbangert) [inline packages/fxa-auth-server/scripts/move-customers-to-new-plan/move-customers-to-new-plan.ts:96]: We already have an index created for this?

Comment (julianpoy) [reply packages/fxa-auth-server/scripts/move-customers-to-new-plan/move-customers-to-new-plan.ts:96]: No, one would need to be created

**Review by bbangert — APPROVED** [review]

**Pull request merged**